components/places/src/api/matcher.rs

```diff
@@ -334,6 +335,28 @@ impl SearchResult {
     }
 }
 
+impl From<SearchResult> for SearchResultMessage {
+    fn from(res: SearchResult) -> Self {
+        Self {
+            url: res.url.into_string(),
+            title: res.title,
+            frecency: res.frecency,
+            reasons: res
+                .reasons
+                .iter()
+                .map(|r| match r {
+                    MatchReason::Keyword => 1,
```

Comment:
Can we use the enum cases instead? Example here:

https://github.com/mozilla/application-services/blob/2649158b61c4e6d7f87bd2e3f16553393094674b/components/fxa-client/src/ffi.rs#L129

Reply:
I mimicked the example above. Hope I got the idea correctly!